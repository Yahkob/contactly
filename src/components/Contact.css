.contact {
  border-bottom: 1px solid #ededed;
  padding: 15px;
}

.contact a {
  color: black;
  text-decoration: none;
}

.email {
  font-style: italic;
  color: #505050bd;
  color: #777;
}

