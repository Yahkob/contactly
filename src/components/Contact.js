import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Contact.css';

const Contact = ({ contact }) => (
  <div className="contact">
    <Link to={contact.id}>
      <div className="name">
        {contact.name}
      </div>
    </Link>
  </div>
);

Contact.Proptypes = {
  contact: PropTypes.object
};

export default Contact;
