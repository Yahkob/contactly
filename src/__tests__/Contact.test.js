import React from 'react';
import Contact from '../components/Contact.js';
import renderer from 'react-test-renderer';
import MockData from '../contacts.json';
import { MemoryRouter } from "react-router-dom";

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Contact contact={MockData[0]} />
      </MemoryRouter>
    ).toJSON();
expect(tree).toMatchSnapshot();
});
