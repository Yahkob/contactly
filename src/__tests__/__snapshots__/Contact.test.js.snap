// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="contact"
>
  <a
    href="1"
    onClick={[Function]}
  >
    <div
      className="name"
    >
      Batman
    </div>
  </a>
</div>
`;
